src/SignalR/common/testassets/Tests.Utils/InProcessTestServer.cs

```diff
@@ -144,12 +147,19 @@ private string RenderLogs(IList<LogRecord> logs)
             return builder.ToString();
         }
 
-        public override void Dispose()
+        public override async ValueTask DisposeAsync()
         {
-            _extraDisposable?.Dispose();
-            _logger.LogInformation("Shutting down test server");
-            _host.Dispose();
-            _loggerFactory.Dispose();
+            try
+            {
+                _extraDisposable?.Dispose();
+                _logger.LogInformation("Start shutting down test server");
+            }
+            finally
+            {
+                await _host.StopAsync();
+                _host.Dispose();
```

Comment:
@davidfowl @Tratcher 

@BrennanConroy and I learned by investigating test failures related to this change that unlike WebHost.Dispose(), the generic Host.Dispose() just disposes all services.

Even when configured with ConfigureWebHost, Host.Dispose() does not fire ApplicationStopping, ApplicationStopped or call IServer.StopAsync(). Instead KestrelServer just gets disposed which completely bypasses graceful shutdown and aborts all connections. Naturally, this lead to DiagnosticMemoryPool errors in the tests.

I think we should make Host.Dispose() gracefully shutdown the server like WebHost.Dispose() did. Should I file an issue for this?

Reply:
Feel free to open an issue for triage, it's out of scope for this PR.

I don't think Dispose should be graceful, that's why we added Stop and helpers like Run that call Stop and Dispose for you. I think this was a conscious choice back in 2.2 or 3.0, if not widely discussed.

Reply:
I was in no way suggesting we change the generic host behavior in this PR. I'm just gauging whether it's worthwhile to even create an issue. It doesn't sound like we're too concerned. Having Dispose() be abortive does make it less sync-over-async which is nice.